Sources/TuistCache/ContentHashing/TargetActionsContentHasher.swift

```diff
@@ -0,0 +1,43 @@
+import Foundation
+import TuistCore
+
+public protocol TargetActionsContentHashing {
+    func hash(targetActions: [TargetAction]) throws -> String
+}
+
+public final class TargetActionsContentHasher: TargetActionsContentHashing {
+    private let contentHasher: ContentHashing
+
+    // MARK: - Init
+
+    public init(contentHasher: ContentHashing = ContentHasher()) {
+        self.contentHasher = contentHasher
+    }
+
+    // MARK: - TargetActionsContentHasher
+
+    /// Returns the hash that uniquely identify an array of target actions
+    /// The hash takes into consideration the content of the script to execute, the input paths, the output paths, the outputFileListPaths, the name of the tool to execute, the order, the arguments and its name
+    public func hash(targetActions: [TargetAction]) throws -> String {
+        var stringsToHash: [String] = []
+        for targetAction in targetActions {
+            let contentHash = try contentHasher.hash(targetAction.path ?? "")
+            let inputPaths = targetAction.inputPaths.map { $0.pathString }
```

Comment:
Here we need to hash the content of the file.

Reply:
Done!